gtdynamics/optimizer/AugmentedLagrangianOptimizer.cpp

```diff
@@ -47,7 +47,8 @@ void update_parameters(const EqualityConstraints& constraints,
 gtsam::Values AugmentedLagrangianOptimizer::optimize(
     const gtsam::NonlinearFactorGraph& graph,
     const EqualityConstraints& constraints,
-    const gtsam::Values& initial_values) const {
+    const gtsam::Values& initial_values,
+    boost::optional<ConstrainedOptResult&> opt_result) const {
```

Comment:
Google style says: in/out parameters should be pointers. Pls use `ConstrainedOptResult*a` and pass in &results…

Reply:
Done